vueui/README.md

```diff
@@ -0,0 +1,180 @@
+```js
+\vueui\webpack.config.js: line 7
+   mode: 'production', // And comment this one out
+   //mode: 'development', // Uncomment this line to set mode to development
+```
+ - Enable debugging inside ui datum. (This will always push new JS file each time open() is called and show data in JSON format at the end of ui)
+```DM
+\code\modules\vueui\ui.dm: line 5
+#define UIDEBUG 0 // Change 0 to 1
+```
+ - Use `\vueui\template.html` in Internet explorer to use inspector to analyze ui behaviour. Also don't forget to copy paste data from actual ui to this debug ui inside templates code. 
+
+## Vue syntax
+You should look at [official Vue.js guide](https://vuejs.org/v2/guide/syntax.html). As it's more detailed and more accurate than any explanation that could have been written here.
+### Data passed to uis
+To access global metadata for this ui, use `this.$root.$data` or `$root.$data`, depending on context. To simplify access you can use following hack that links global data to local component data. It simplifies access to data. following explanation assumes you have done so. _Note: changes made on client side aren't sent to server, so **please** do not alter them._
+```vue
+<script>
+export default {
+    name: 'view-name',
+    data() {
+        return this.$root.$data;
+    }
+};
+</script>
+```
+#### `state` - Representation of `ui.data` inside ui.
+This is outside metadata, so changes to this variable can be made, but this variable is expected to be a object or how BYOND calls it - a keyed list, using other type *will* cause errors. This variable is main way to interact with server without using `Topic()`.
+
+Note: `state` can get very out of sync with `ui.data`, this often occur on rapid changes. So to make sure latest data gets to ui datum, when ui data is needed for usage, `<vui-button>` parameter `push-state` should be set. Example: `<vui-button push-state :params="{copy: 1}">Copy</vui-button>` in photocopier.vue - makes sure that copy amount is up to date. When `state` object is huge, it's discouraged to solve this issue like this. Then it should be solved by sending needed data inside `params`.
+#### `wtime` - Global world time since boot, client side guess of `word.time`
+This is constantly counting counter updated every 200ms representing time since server has started. This should be used for displaying counters, timers, as it doesn't depend on pushed state so much, so it allows making better user experiences.
+#### `title` - This ui's title, copy of `ui.title`
+This uis title, mostly used by header components.
+#### `status` - Interactivity status, mirror of `ui.status`
+Topic status, used to determine how interactive is ui. Meanings of numbers can be seen `\code\__defines\machinery.dm`.
+#### `assets` - This makes `add_asset`, `remove_asset` and `<vui-img>` tick
+As described, this makes them tick. This ***should*** shouldn't be used, unless you know what you are doing.
+#### `active` - Representation of `ui.activeui`
```

Comment:
What exactly does this var do?

Reply:
It mostly tells ui what template should be used.